.github/workflows/ci.yml

```diff
@@ -16,28 +16,44 @@ jobs:
       fail-fast: false
       matrix:
         ruby: [2.7, '3.0', 3.1, 3.2, 3.3, jruby-9.4]
-        gemfile: [rails-7.0.gemfile, rails-7.1.gemfile, rails-main.gemfile]
+        rails: ['7.0', 7.1, main]
 
         exclude:
           # Rails 7.1 dropped support for older rubygems
-          - gemfile: rails-main.gemfile
+          - rails: main
             ruby: 2.7
-          - gemfile: rails-main.gemfile
+          - rails: main
             ruby: 3.0
+          # JDBC adapters don't support the latest Rails
+          - rails: main
+            ruby: jruby-9.4
 
+    env:
+      RAILS_ENV: test
+      RAILS_VERSION: ${{ matrix.rails }}
     steps:
       - name: Setup Ruby
         uses: ruby/setup-ruby@v1
         with:
           ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
 
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+
       - name: Install gems
-        run: bundle install --gemfile=gemfiles/${{ matrix.gemfile }}
+        run: bundle install
+        env:
+          BUNDLE_GEMFILE: gemfiles/rails-${{ matrix.rails }}.gemfile
```

Comment:
was the workspace path not working?

Reply:
Yeah, I think that was causing the issue? I can try again now that I got everything green and see what was breaking.

Reply:
huh, weird

Reply:
not a big deal